compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -913,12 +924,18 @@ fn transform_ty<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, options: TransformTyOptio
             );
         }
 
+        ty::Alias(alias_kind, alias_ty) => {
```

Comment:
This type shouldn't be encounterable... it should really be normalized away. That is, `<i32 as Baz>::Foo` is exactly `i32` at codegen time.

Reply:
You may need to normalize your types with `tcx.normalize_erasing_regions(..)` using `ty::ParamEnv::reveal_all()`

Reply:
Humm... for CFI purposes `foo(_: Type15)` should be grouped separately from `bar(_: i32)`, but `Type15` is a `repr(transparent)` type, so it's actually `foo(_: <i32 as Baz>::Foo)`, which is equivalent to `foo(_: i32)` and `foo(_: Type15)` will accept an `i32` as argument. Do you think there is any case that is different than this `repr(transparent)` case where functions actually should by grouped separately? If we were to substitute, is it possible for an `AliasTy` to have more than one substitution?

Reply:
Changed as you suggested. Let me know if it looks good to you! My only concern of normalizing ty::Alias before encoding is whether there is any case where we would want to group functions separately by opaque and projection types being used as return and parameter types, but I don't immediately see any such case.

Reply:
Sorry, you're right. They are just transformed before encoding for CFI purposes (see https://github.com/rust-lang/rust/blob/master/compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs#L803), but still require an explicit type conversion. I was referring to CFI accepting an i32 as argument.